### PR TITLE
chore: upgrade `flair` to 0.12

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

As older versions of `flair` (< 0.12) were having problems during the installation, I've decided to upgrade `flair` to 0.12 and pin it. Besides the package upgrade, some things had to be updated in the `FlairMonitor` for the unit tests to pass when installing the latest version of `flair`.

Some more information about the installation issue at https://github.com/flairNLP/flair/issues/3129

**Type of change**

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

- [x] Updated the `FlairMonitoring` unit tests

**Checklist**

- [X] I have merged the original branch into my forked branch
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/argilla/monitoring/_flair.py [hunk 1]
```diff
@@ -63,7 +65,7 @@ def predict(self, sentences: Union[List[Sentence], Sentence], *args, **kwargs):
         metadata = kwargs.pop("metadata", None)
         result = self.__model__.predict(sentences, *args, **kwargs)
 
-        if isinstance(sentences, Sentence):
+        if not isinstance(sentences, list):
```
Comment (frascuchon) [inline src/argilla/monitoring/_flair.py:68]: Be aware of using this condition only will match for `list`. If you're using a generator or other kinds of sequences to provide a sentence batch, the `predict` wrapper will fail. Even if the original method (`model.predict`) is working

```python
instance((i for i in [1,2,3]), list) # False
```

Comment (alvarobartt) [reply src/argilla/monitoring/_flair.py:68]: Even though you're right, I was following the type-hint for the `sentences` arg, that's why I kept that check as `isinstance(..., list)`, as I guess we're only expecting either a list of `Sentence` or a single `Sentence`.

**Review by gabrielmbmb — APPROVED** [review]
LGTM!

**Review by davidberenstein1957 — APPROVED** [review]
LGTM, but maybe the SetFit updates were supposed to go to another PR?

**Comment by alvarobartt** [comment]
> LGTM, but maybe the SetFit updates were supposed to go to another PR?

Those were indeed part of another PR, see https://github.com/argilla-io/argilla/pull/3015, those are the commits from the merge 😄 (not sure why those were rebased anyway)

**Pull request merged**